src/coord/src/coord.rs

```diff
@@ -2218,26 +2269,40 @@ where
                         CatalogItem::Table(_) => {
                             self.report_table_update(id, *schema_id, &name.item, 1)
                         }
+                        CatalogItem::Source(_) => {
+                            self.report_source_update(id, *schema_id, &name.item, 1);
+                        }
+                        CatalogItem::Sink(_) => {
+                            self.report_sink_update(id, *schema_id, &name.item, 1);
+                        }
                         _ => (),
                     }
                 }
                 catalog::OpStatus::UpdatedItem {
                     schema_id,
                     id,
                     from_name,
+                    to_name,
                     item,
                 } => {
                     // Remove old name from mz_catalog_names.
                     self.report_catalog_update(*id, from_name.to_string(), -1);
 
                     // Add new name to mz_catalog_names.
-                    let updated_name = self.catalog.humanize_id(expr::Id::Global(*id)).unwrap();
-                    self.report_catalog_update(*id, updated_name.clone(), 1);
+                    self.report_catalog_update(*id, to_name.to_string(), 1);
 
                     // Remove old name and add new name to relevant mz system tables.
-                    if let CatalogItem::Table(_) = item {
-                        self.report_table_update(&id, *schema_id, &from_name.item, -1);
-                        self.report_table_update(&id, *schema_id, &updated_name, 1);
+                    match item {
+                        CatalogItem::Source(_) => {
+                            self.report_source_update(id, *schema_id, &from_name.item, -1);
+                            self.report_source_update(id, *schema_id, &to_name.item, 1);
+                        }
+                        CatalogItem::Sink(_) => {
+                            self.report_sink_update(id, *schema_id, &from_name.item, -1);
+                            self.report_sink_update(id, *schema_id, &to_name.item, 1);
+                        }
+                        CatalogItem::Table(_) => unreachable!(), // Table names can't be altered.
```

Comment:
Oh, hm, maybe we should fix that. I don't see any reason why we can't rename tables?

Reply:
Not for this PR though! #4234